### PR TITLE
Validation for sign up and login

### LINKED ISSUE #123: front-end validation

### PR DESCRIPTION
**Pull request by amanshawar**
relate #123 
fix #123

### CODE REVIEW

File: client/src/components/Login/index.js [hunk 1]
```diff
@@ -22,19 +23,54 @@ class Login extends Component {
...
+    const errors = {
+      usernameError: '',
+      passwordError: '',
+      isErrorUsername: false,
+      isErrorPassword: false,
+    };
+    if (this.state.username < 1) {
+      isError = true;
+      errors.isErrorUsername = true;
+      errors.usernameError = 'please enter Username';
```
Comment (jema28) [inline client/src/components/Login/index.js]: Please enter your username.

File: client/src/components/Login/index.js [hunk 2]
```diff
@@ -22,19 +23,54 @@ class Login extends Component {
...
+    }
+    if (this.state.password < 1) {
+      isError = true;
+      errors.isErrorPassword = true;
+      errors.passwordError = 'please enter Password';
```
Comment (jema28) [inline client/src/components/Login/index.js]: Please enter your password.

File: client/src/components/Login/index.js [hunk 3]
```diff
@@ -22,19 +23,54 @@ class Login extends Component {
...
+  login = () => {
+    const err = this.validate();
+    if (!err) {
+      const { history } = this.props;
+      const inputs = { username: this.state.username, password: this.state.password };
+      axios.post('/login', inputs).then(({ data }) => {
+        if (data.success) {
+          history.push('/item-list');
+        } else {
+          this.setState({ passwordError: 'the Username and Password not match', isErrorUsername: true, isErrorPassword: true });
```
Comment (jema28) [inline client/src/components/Login/index.js]: Username or password is incorrect.

File: client/src/components/SignUp/signup.style.js [hunk 4]
```diff
@@ -52,16 +64,13 @@ export const StyledText = styled.div`
...
-  padding:7px 0px;
-  @media (max-width: 650px) {
-    font-size: 16px;
-  }
-  ::placeholder{
-    font-family:'Assistant';
-  }
+
+export const ErrorMessage = styled.h1`
+font-size: 16px;
```
Comment (jema28) [inline client/src/components/SignUp/signup.style.js]: Indentation habibti!

File: server/controllers/signup.js [hunk 5]
```diff
@@ -51,7 +51,7 @@ exports.signUp = (req, res) => {
           },
         );
       } else {
-        return res.json({ error: 'the password and confirm password not match' });
+        return res.json({ success: false, error: 'the password and confirm password not match' });
```
Comment (jema28) [inline server/controllers/signup.js:54]: Your username or password is incorrect.

File: server/controllers/signup.js [hunk 6]
```diff
@@ -14,7 +14,7 @@ exports.signUp = (req, res) => {
   User.findOne({ username })
     .then((user) => {
       if (user) {
-        return res.json({ error: 'User exist , please use another one' });
+        return res.json({ success: false, error: 'User exist , please use another one' });
```
Comment (jema28) [inline server/controllers/signup.js:17]: This username already exists.

File: client/src/components/SignUp/index.js [hunk 7]
```diff
@@ -14,32 +14,78 @@ import {
...
+      } else if (this.state.username < 1) {
+        isError = true;
+        errors.isErrorUsername = true;
+        errors.usernameError = 'Username is required';
       }
+    }
+    if (this.state.password !== this.state.confirmPassword) {
+      isError = true;
+      errors.isErrorPassword = true;
+      errors.passwordError = 'The password and confirm password not match';
```
Comment (jema28) [inline client/src/components/SignUp/index.js]: Passwords do not match.

File: client/src/components/SignUp/index.js [hunk 8]
```diff
@@ -14,32 +14,78 @@ import {
...
-    axios.post('/signup', inputs).then(({ data }) => {
-      if (data.success) {
-        history.push('/login-form');
-      } else {
-        history.push('/error');
+    for (let i = 0; i < characters.length; i++) {
+      if (this.state.username.includes(characters[i])) {
+        isError = true;
+        errors.isErrorUsername = true;
+        errors.usernameError = 'Username could contains letter,number and \'-\',\'_\' just';
```
Comment (jema28) [inline client/src/components/SignUp/index.js]: Username should only contain letters, numbers, underscores and dashes.

File: client/src/components/Login/login.style.js [hunk 9]
```diff
@@ -59,10 +59,20 @@ export const StyledInput = styled.input`
...
+  border-bottom: ${props => (props.StyleError ? '1px solid red' : '1px solid #7D7D7D')};
   @media (max-width: 650px) {
     font-size: 16px;
   }
   ::placeholder{
     font-family:'Assistant';
   }
 `;
+export const ErrorMessage = styled.h1`
+font-size: 16px;
```
Comment (jema28) [inline client/src/components/Login/login.style.js]: Indentation habibti!

**Review by jema28 — CHANGES_REQUESTED** [review]

File: server/controllers/signup.js (hunk 5 above)
Comment (amanshawar) [reply server/controllers/signup.js:54]: here

Comment (jema28) [reply server/controllers/signup.js:54]: Passwords do not match.

**Review by Hadeel-Salamin — APPROVED** [review]

**Review by jema28 — APPROVED** [review]

**Pull request merged**